lms/djangoapps/certificates/api.py

```diff
@@ -920,3 +922,34 @@ def _has_passed_or_is_allowlisted(course, student, course_grade):
     has_passed = course_grade and course_grade.passed
 
     return has_passed or is_allowlisted
+
+
+def invalidate_certificate(user_id, course_key_or_id, source):
```

Comment:
[question] Is it course_key_or_id because it is course id in some places? In the code here, I see it is course_key, so wasn't sure if it's just different in different places. And if so, I'm guessing that once the legacy piece is deprecated, we would only be using one of these terms? Otherwise, I'm wondering if there would need to be a separate refactoring effort for consistency.

Reply:
To my understanding, there's an initiative to move from just using strings as "course ID's" to "Course Keys", which are essentially a Django object that contains the course ID + some other stuff. I _believe_ that the course_key_or_id variable name is used commonly throughout edx-platform because we want to account for the fact that, because we haven't migrated every instance of a course ID string to a Course Key object, that we might end up using one or the other.